Sources/SwiftProtobuf/ExtensionMap.swift

```diff
@@ -0,0 +1,38 @@
+// Sources/SwiftProtobuf/ExtensionMap.swift - Extension support
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// A set of extensions that can be passed into deserializers
+/// to provide details of the particular extensions that should
+/// be recognized.
+///
+// -----------------------------------------------------------------------------
+
+/// A collection of extension objects.
+///
+/// An `ExtensionMap` is used during decoding to look up
```

Comment:
I'm fine with the name, but I'll float another possible name: ExtensionRegistry.